.github/workflows/android.yml

```diff
@@ -1,32 +1,58 @@
 name: Build Android and upload to Play Store
 on: 
-  push: 
-    branches: [ main,pre-main ]
+  workflow_call:
+    secrets:
+      CODESIGNING_DEPLOY_KEY:
+        required: true
+      GOOGLE_PLAY_JSON_KEY:
+        required: true
+      MATCH_PASSWORD:
+        required: true
+
 jobs:
-  build-ios: 
+
+  build-droid: 
     name: Build Flutter Android
     runs-on: ubuntu-20.04
     steps:
+      - name: Set environment variables
+        run: |
+          if [[ "${{ github.ref }}" == "refs/heads/main" ]]; then
+            echo "Running on the main branch, deploying to production."
+            echo "export package_name=net.givtapp.kids" >> $GITHUB_ENV
+            echo "export flavor=production" >> $GITHUB_ENV
+            echo "export track=production" >> $GITHUB_ENV
+          else
+            echo "Running on a different branch, deploying to test."
+            echo "export package_name=net.givtapp.kids.test" >> $GITHUB_ENV
+            echo "export flavor=development" >> $GITHUB_ENV
+            echo "export track=internal" >> $GITHUB_ENV
+          fi
+
       - name: Checkout
         uses: actions/checkout@v3
+
       - name: Setup Ruby
         uses: ruby/setup-ruby@v1.118.0
         with: 
           ruby-version: '3.0'
           bundler-cache: true
+
       - name: Auth codesigning
         uses: webfactory/ssh-agent@v0.5.4
         with:
-          ssh-private-key: ${{ secrets.CODESIGNING_DEPLOY_KEY }}          
+          ssh-private-key: ${{ secrets.CODESIGNING_DEPLOY_KEY }}
+
       - name: Install Flutter
         uses: subosito/flutter-action@v2
         with:
-          channel: 'stable'          
+          channel: 'stable'
+
       - name: Run Fastlane
         uses: maierj/fastlane-action@v2.3.0
         with:
           lane: android build_and_deploy
           subdirectory: android
         env:
-          GOOGLE_PLAY_JSON_KEY_DATA: ${{ secrets.GOOGLE_PLAY_JSON_KEY_DATA }}
+          GOOGLE_PLAY_JSON_KEY_DATA: ${{ secrets.GOOGLE_PLAY_JSON_KEY }}
```

Comment:
How did solved this in the end? Did you use the key that we have store previously?

Reply:
@alexbejann Maikel doesn't have access either, so he can't give me access..
We have no choice atm :(

Reply:
Sounds like we would have to address this in the upcoming days